src/cow_string.hpp

```diff
@@ -103,20 +105,21 @@ class cow_string {
 
     // Moves the contents and invalidates the string if the buffer has been
     // modified, otherwise it does nothing
```

Comment:
The comment does not apply. It always invalidates the string. In fact, the conversion operator should be called on rvalue references only, so that the compiler can warn on uses after move.